### PR TITLE
wait for build succeeded before launch debug session

### PR DESCRIPTION
**Pull request by SummerSun**
Should wait for build succeeded before launch debug session.

Un-comment the ```// only rebuild for single file case before the build error issue is resolved.``` for no detailed issue information was tracked and could not repro the issue either.

### CODE REVIEW

**Comment by SummerSun** [comment]
Hold this PR for random build error from language server side even the project is all good.

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -94,7 +94,13 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                     // for opened with folder, return directly.
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Totally don't understand the logic at `L92 ~ L96`, can you help update the comment?

File: src/configurationProvider.ts [hunk 2]
```diff
@@ -88,13 +88,19 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
...
+            if (Object.keys(config).length === 0 && !folder) {
+                config.type = "java";
+                config.name = "Java Debug";
+                config.request = "launch";
+            }
+
+            if (config.request === "launch") {
                 try {
                     const buildResult = await vscode.commands.executeCommand(commands.JAVA_BUILD_WORKSPACE, false);
                     console.log(buildResult);
```
Comment (andxu) [inline src/configurationProvider.ts]: please remove the console.log() but the logic to check build status with the right kind of error message.

File: src/configurationProvider.ts [hunk 3]
```diff
@@ -88,29 +88,27 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
...
             }
 
-            if (Object.keys(config).length === 0) { // No launch.json in current workspace.
-                // check whether it is opened as a folder
-                if (folder !== undefined) {
-                    // for opened with folder, return directly.
-                    return config;
-                }
-                // only rebuild for single file case before the build error issue is resolved.
+            // just create and open the lanuch.json if there is no lanch.json in current folder
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Change the comment to `If no launch.json exists in the current workspace folder, delegate to provideDebugConfigurations api to generate the initial launch.json configurations.`

File: src/configurationProvider.ts [hunk 4]
```diff
@@ -88,29 +88,27 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
...
+            if (Object.keys(config).length === 0 && folder !== undefined) {
+                return config;
+            }
+
+            // generate config in memory if there is noly single file and no launch.json
```
Comment (testforstephen) [inline src/configurationProvider.ts]: If it's the single file case that no workspace folder is opened, generate debug config in memory.

**Comment by SummerSun** [comment]
#207

**Pull request merged**